pkg/querier/ingester_streaming_queryable.go

```diff
@@ -47,8 +47,8 @@ func (i ingesterQueryable) Get(ctx context.Context, userID string, from, through
 		return nil, promql.ErrStorage{Err: err}
 	}
 
-	chunks := make([]chunk.Chunk, 0, len(results))
-	for _, result := range results {
+	chunks := make([]chunk.Chunk, 0, len(results.Chunkseries))
```

Comment:
I manually tested this PR and querying the ingesters does not work. I think it's because `results.Timeseries` is not handled here in the `ingesterQueryable.Get()`.

_We also need better unit tests coverage here, to ensure this issue is covered too._

Reply:
Yea you're right. An unfortunate result from merged the results and trying to reuse the `ingesterStreamingQueryable` as common code. It actually looks like the `Select` function is never utilized since `newUnifiedChunkQueryable.Select` calls `Get` instead of `Select`. 

So to resolve that I'll have to convert the `client.TimeSeries` into chunks, which is what I was trying to avoid.

Reply:
@pstibrany Do you have thoughts about this? Looks unfortunate if we have to convert timeseries into chunks. I haven't deeply looked into alternatives, so I'm asking if you have any idea.

Reply:
This is only used by chunks-based querier to unify access to different chunk sources. TSDB based querier doesn't use this. We only convert chunk-based series into `[]chunk.Chunk` here.

Reply:
Ignore my comment pls, it's clearly wrong.

Reply:
I think with current design of Querier, it's difficult to avoid copying and adapting to expected return value (We already do the same conversion in BlockQuerier). Long-term, we should refactor querier to be based on Queryables more than chunk stores.